File: examples/C++/MyProjects/Doodle/step02/step02.cc
// ============================================================================
// step02.cc: Handling events in the main window with Qt's signals/slots
//
// Tutorial - Qt Scribble Application
// Based on a similar tutorial for Borland ObjectWindows Library (OWL)
//
// @author Manish Bhobé for Nämostuté Ltd.
// My experiments with C++,Qt, Python & PyQt.
// Code is provided for illustration purposes only! Use at your own risk.
// =============================================================================

#include "DrawWindow.h"
#include "chocolaf.h"
#include <QApplication>
#include <QTextStream>
#include <QtGui>

QTextStream cout(stdout, QIODeviceBase::WriteOnly);

int main(int argc, char **argv)
{
  QApplication app(argc, argv);
  Chocolaf::setChocolafStyle(app, "WindowsDark");

  QStringList args = QCoreApplication::arguments();
  foreach (auto arg, args)
    cout << arg << " ";
  cout << Qt::endl;

  app.setApplicationName(app.translate("main", "Qt Scribble"));

  // create the GUI
  DrawWindow mainWindow;
  Chocolaf::centerOnScreenWithSize(mainWindow, 0.75, 0.75);
  mainWindow.show();

  return app.exec();
}

File: examples/Python/MyProjects/PyDoodle/step02/step02.py
"""
// ============================================================================
// step02.py: handling events in the main window
//      the main window responds to left & right mouse clicks in the client
//      area with message boxes and also the close event from OS
//
// Tutorial - PyQt5 Doodle Application
// Based on a similar tutorial for Borland(R) ObjectWindows(TM) Library (OWL)
// @author: Manish Bhobé
// My experiments with the Qt Framework. Use at your own risk!!
// ============================================================================
"""

import sys
import platform

import qtpy

from mainWindow import *


def main():
    app = QApplication(sys.argv)
    app.setStyle("Fusion")
    print(
        f"PyQt Doodle - running with Python {platform.python_version()}, "
        + f"Qt {qtpy.QT_VERSION}, PyQt {qtpy.PYQT_VERSION} on {platform.system()}"
    )

    mainWindow = MainWindow()
    mainWindow.show()

    sys.exit(app.exec())


if __name__ == "__main__":
    main()
